tal/render/render.py

```diff
@@ -1,443 +1,707 @@
+                        f"{self.experiment_name}_{render_name}.log",
+                    ),
+                    "w",
+                )
             # NOTE: something here has a memory leak (probably Mitsuba-related)
             # We run Mitsuba in a separate process to ensure that the leaks do not add up
             # as they can fill your RAM in exhaustive scans
-            run_mitsuba_f = partial(mitsuba_backend.run_mitsuba, nlos_scene_xml, hdr_path, defines,
-                                    render_name, sys.stdout, args, queue=queue)
-            if os.name == 'nt':
-                # NOTE: Windows does not support multiprocessing
-                run_mitsuba_f()
-            else:
-                process = multiprocessing.Process(target=run_mitsuba_f)
-                try:
-                    process.start()
-                    process.join()
-                except KeyboardInterrupt:
-                    process.terminate()
-                    raise KeyboardInterrupt
-            if scan_type == 'exhaustive' and i == 0:
-                size_bytes = os.path.getsize(hdr_path)
-                final_size_gb = size_bytes * len(laser_lookats) / 2**30
-                pbar.set_description(
-                    f'Rendering {experiment_name} ({scan_type}, estimated size: {final_size_gb:.2f} GB)...')
-            if args.do_logging and not args.dry_run:
-                past_elems = []
-                while not queue.empty():
-                    e = queue.get()
-                    past_elems.append(e)
-                    if isinstance(e, Exception):
-                        log(LogLevel.ERROR, '')
-                        log(LogLevel.ERROR, '/!\ Mitsuba thread got an exception!')
-                        log(LogLevel.ERROR, '')
-                        for pe in past_elems[:-1]:
-                            log(LogLevel.ERROR, pe)
-                        log(LogLevel.ERROR, '')
-                        raise e
-                    else:
-                        logfile.write(e)
-                    logfile.flush()
-                queue.close()
-                logfile.close()
-
-        if args.dry_run:
-            return
+            queue = StdoutQueue()
+            run_mitsuba_f = partial(
+                self.mitsuba_backend.run_mitsuba,
+                self.steady_scene_xml_path,
+                hdr_path,
+                dict(),
+                render_name,
+                logfile,
+                self.args,
+                sensor_index,
+                queue,
+            )
+            self._run_mitsuba(hdr_path, ldr_path, logfile, queue, run_mitsuba_f)
+
+    def _run_mitsuba(self, hdr_path, ldr_path, logfile, queue, run_mitsuba_f):
+        if platform.system() == "Windows":
+            # NOTE: Windows does not support multiprocessing
+            run_mitsuba_f()
+        else:
+            # HACK: on macOS "spawn" method, which is the default since 3.8,
```

Comment:
AFAIK you are the first macOS user of tal so it's nice to know that it works. Windows gave some problems to other guys in our lab